# Seattle Staff Demographics Dataset
June 

9/23/19

## Abstract

In this lab we want to be able to tell a story with the dataset we downloaded. More specifically, the story between the Age and Hourly Rate data with the City of Seattle Staff Demographics Dataset (from https://data.seattle.gov/City-Business/City-of-Seattle-Staff-Demographics/5avq-r9hj). This dataset is important to analyze because it shows the people who work for the City of Seattle and their ages and how much they are paid. From this information, we will be able to draw engaging questions about this particular dataset.   

## Dataset Preparation

In [4]:
#open the csv file
seattle_file = open("City_of_Seattle_Staff_Demographics.csv", "r");

#make a empty list to store the data from Dataset
seattle_data = [];

#put all lines from the seattle_file into a list
for row in seattle_file:
    seattle_data.append(row);
    
#close seattle_file
seattle_file.close();

print(seattle_data[:5]);


['Race/Ethnicity,Sex,Department,Age,Hourly Rate,Regular/Temporary,Employee Status\n', 'Hispanic or Latino,M,Municipal Court,57,29.75,R,A\n', 'Hispanic or Latino,F,Municipal Court,39,27.18,R,A\n', 'Hispanic or Latino,M,Police Department,52,59.75,R,A\n', 'Hispanic or Latino,M,Fire Department,30,38.28,R,A\n']


In [29]:
#remove header
#import data from seattle_file
import statistics

del seattle_data[0];

# make two empty lists - one for age and one for wage seattle data
# make a list of just the age and just the wage data
age = [];
wage = [];  

# loop (go through every row one by one) through the seattle_data
for item in range(len(seattle_data)):

    
    # what this line is doing is splitting out each comma separated item from a row and turning it into a list 
    # After being split, the three separate strings are then stored in another list called seattle_row
    seattle_row = seattle_data[item].split(",")
    
    #now that the row is split into three parts, we can pick and choose 
    # add the second and the third to specific lists for age and wage
    age.append(int(seattle_row[3]))
    wage.append(float(seattle_row[4]))
    
agemin = min(age)

agemax = max(age)

wagemax = max(wage)

wagemin = min(wage)

print("age minimum:")
print(agemin)
print("age maximum:")
print(agemax)
print("wage maximum:")
print(wagemax)
print("wage mimimum:")
print(wagemin)


age minimum:
14
age maximum:
92
wage maximum:
162.8353
wage mimimum:
5.53


## Data Modeling

In [5]:
#use the function len to count how many items in list
someNbrs = age
count = len(someNbrs)
print("Nbr of values is ", count)

# to add up all the items in the list, use the function sum
total = sum(someNbrs)
print("Total sum of values is ", total)

# you can then use these two values to calculate the average (also called the mean)
avg=total/count
print("Average of values is ", avg)

# or, use the function statistics.mean to do the calculations
# you must include the import statistics line first
import statistics
another_mean = statistics.mean(someNbrs)
print("Mean (average) of values is", another_mean)

NameError: name 'age' is not defined

In [3]:
# to count how many items are in the list, use the function len
someNbrs = wage
count = len(someNbrs)
print("Nbr of values is ", count)

# to add up all the items in the list, use the function sum
total = sum(someNbrs)
print("Total sum of values is ", total)

# you can then use these two values to calculate the average (also called the mean)
avg=total/count
print("Average of values is ", avg)

# or, use the function statistics.mean to do the calculations
# you must include the import statistics line first
import statistics
another_mean = statistics.mean(someNbrs)
print("Mean (average) of values is", another_mean)

NameError: name 'wage' is not defined

## Data Analysis and Conclusion

In conclusion, the oldest person working is 92 years old and the youngest is 14 years old. The highest paid person is paid $162.84 while the least paid person was paid $5.53 per an hour. 

## Acknowledgements

In [ ]:
Ms. Sconyers: 
- Thank you for letting me not only copy and use your codes, but for explaining what to do for each part of this lab.
    
Ziah:
- Thank you for helping organize my data, helping with my Data Preparation and Modeling while also checking
my lab over and helping with the errors.
Navya: Thank you for answering any of my questions and explaining the Abstract to me
Mark: Thank you for teaching me how to open the csv file and looking over mine
Rihan: Thank you for walking me through how to do the analysis part.